x-pack/plugins/reporting/public/share_context_menu/csv_export_modal.tsx

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiCopy,
+  EuiForm,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from '@elastic/eui';
+import type { IUiSettingsClient, ThemeServiceSetup, ToastsSetup } from '@kbn/core/public';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n-react';
+import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+import type { BaseParams } from '@kbn/reporting-common/types';
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import useMountedState from 'react-use/lib/useMountedState';
+import url from 'url';
+import { ReportingAPIClient } from '../lib/reporting_api_client';
+import { ErrorUrlTooLongPanel, ErrorUnsavedWorkPanel } from './reporting_panel_content/components';
+import { getMaxUrlLength } from './reporting_panel_content/constants';
+
+export interface CsvModalProps {
+  apiClient: ReportingAPIClient;
+  toasts: ToastsSetup;
+  uiSettings: IUiSettingsClient;
+  reportType: string;
+  requiresSavedState: boolean; // Whether the report to be generated requires saved state that is not captured in the URL submitted to the report generator.
+  getJobParams: (forShareUrl?: boolean) => Omit<BaseParams, 'browserTimezone' | 'version'>;
```

Comment:
```suggestion
  getJobParams: JobAppParamsCSV;
```

JobAppParamsCSV can be imported with:
`import { JobAppParamsCSV } from '@kbn/reporting-export-types-csv-common';`